### PR TITLE
[3.9] Wrong default value [SQL ]

### PR DESCRIPTION
**Pull request by amitranjan2**
Pull Request for Issue # .

### Summary of Changes

`INT DEFAULT value changed to numeric from char

### Testing Instructions

see the code 
installation/sql/mysql/joomla.sql

installation/sql/postgresql/joomla.sql

### Expected result

same as before

### Actual result
same as before

### Documentation Changes Required
NO

### CODE REVIEW

**Comment by amitranjan2** [comment]
@alikon some days ago  We solved this for joomla [4.0]
remembered?

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -695,18 +695,18 @@ CREATE TABLE IF NOT EXISTS `#__fields` (
...
   `params` text NOT NULL,
   `fieldparams` text NOT NULL,
   `language` char(7) NOT NULL DEFAULT '',
   `created_time` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `created_user_id` int(10) unsigned NOT NULL DEFAULT '0',
+  `created_user_id` int(10) unsigned NOT NULL DEFAULT 0,
   `modified_time` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `modified_by` int(10) unsigned NOT NULL DEFAULT '0',
+  `modified_by` int(10) unsigned NOT NULL DEFAULT 0,
   `access` int(11) NOT NULL DEFAULT '1',
```
Comment (SharkyKZ) [inline installation/sql/mysql/joomla.sql]:
```suggestion
  `access` int(11) NOT NULL DEFAULT 1,
```

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -742,16 +742,16 @@ CREATE TABLE IF NOT EXISTS `#__fields_groups` (
...
+  `ordering` int(11) NOT NULL DEFAULT 0,
   `params` text NOT NULL,
   `language` char(7) NOT NULL DEFAULT '',
   `created` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `created_by` int(10) unsigned NOT NULL DEFAULT '0',
+  `created_by` int(10) unsigned NOT NULL DEFAULT 0,
   `modified` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-  `modified_by` int(10) unsigned NOT NULL DEFAULT '0',
+  `modified_by` int(10) unsigned NOT NULL DEFAULT 0,
   `access` int(11) NOT NULL DEFAULT '1',
```
Comment (SharkyKZ) [inline installation/sql/mysql/joomla.sql]:
```suggestion
  `access` int(11) NOT NULL DEFAULT 1,
```

File: installation/sql/postgresql/joomla.sql [hunk 3]
```diff
@@ -751,16 +751,16 @@ CREATE TABLE "#__fields_groups" (
...
+  "ordering" integer DEFAULT 0 NOT NULL,
   "params" text DEFAULT '' NOT NULL,
   "language" varchar(7) DEFAULT '' NOT NULL,
   "created" timestamp without time zone DEFAULT '1970-01-01 00:00:00' NOT NULL,
-  "created_by" bigint DEFAULT '0' NOT NULL,
+  "created_by" bigint DEFAULT 0 NOT NULL,
   "modified" timestamp without time zone DEFAULT '1970-01-01 00:00:00' NOT NULL,
-  "modified_by" bigint DEFAULT '0' NOT NULL,
+  "modified_by" bigint DEFAULT 0 NOT NULL,
   "access" bigint DEFAULT '1' NOT NULL,
```
Comment (SharkyKZ) [inline installation/sql/postgresql/joomla.sql]:
```suggestion
  "access" bigint DEFAULT 1 NOT NULL,
```

File: installation/sql/mysql/joomla.sql [hunk 4]
```diff
@@ -1702,12 +1702,12 @@ CREATE TABLE IF NOT EXISTS `#__privacy_requests` (
 
 CREATE TABLE IF NOT EXISTS `#__privacy_consents` (
   `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
-  `user_id` int(10) unsigned NOT NULL DEFAULT '0',
+  `user_id` int(10) unsigned NOT NULL DEFAULT 0,
   `state` INT(10) NOT NULL DEFAULT '1',
```
Comment (SharkyKZ) [inline installation/sql/mysql/joomla.sql]:
```suggestion
  `state` INT(10) NOT NULL DEFAULT 1,
```

**Comment by SharkyKZ** [comment]
There are more columns remaining that should be changed, e.g. https://github.com/joomla/joomla-cms/blob/5200e186a45918626f0028348337e65426ee01e4/installation/sql/mysql/joomla.sql#L1654

**Comment by amitranjan2** [comment]
> There are more columns remaining that should be changed, e.g.
> 
> [joomla-cms/installation/sql/mysql/joomla.sql](https://github.com/joomla/joomla-cms/blob/5200e186a45918626f0028348337e65426ee01e4/installation/sql/mysql/joomla.sql#L1654)
> 
> Line 1654 in [5200e18](/joomla/joomla-cms/commit/5200e186a45918626f0028348337e65426ee01e4)
> 
>  `language_client_id` tinyint(3) NOT NULL DEFAULT '1',

done, thanks

**Comment by alikon** [comment]
you should check the update sql files as well 
for example https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-05-02.sql#L8

**Comment by alikon** [comment]
please take a look at all sql updates files 
another example https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-05-24.sql#L6

**Comment by amitranjan2** [comment]
> please take a look at all sql updates files
> another example https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-05-24.sql#L6

Already done
yes, I have checked for mysql as well as  PostgreSQL also, thanks
please tell me if any left

**Comment by alikon** [comment]
this one https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-08-12.sql   for example

**Comment by amitranjan2** [comment]
> this one https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-08-12.sql for example

oops sorry , done

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 5528cd7a7c823395d6e755c66d1f037842f7142b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24058">issues.joomla.org/tracker/joomla-cms/24058</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5528cd7a7c823395d6e755c66d1f037842f7142b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24058">issues.joomla.org/tracker/joomla-cms/24058</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24058">issues.joomla.org/tracker/joomla-cms/24058</a>.</sub>

**Comment by amitranjan2** [comment]
@wilsonge if no issues , please kindly merge  this

**Comment by brianteeman** [comment]
@amitranjan2 patience is a virtue

**Comment by HLeithner** [comment]
thx

**Comment by amitranjan2** [comment]
thanks

**Pull request merged**